### ISSUE TITLE
[rulename] issue title

### ISSUE THREAD

**Issue by grosch** [typescript-eslint/typescript-eslint#1986]
**Repro**

```JSON
{
    "parser": "@typescript-eslint/parser",
    "parserOptions": {
        "ecmaVersion": 2020,
        "sourceType": "module"
    },
    "plugins": [
        "@typescript-eslint"
    ],
    "extends": [
        "plugin:@typescript-eslint/recommended"
    ],
    "rules": {
        "semi": [
            2,
            "never"
        ],
	"quotes": [ "error", "single", { "avoidEscape": true } ]
    }
}
```

```TS
@Component({
    providers: [
        {
            provide: NG_VALUE_ACCESSOR,
            // eslint-disable-next-line @typescript-eslint/no-use-before-define
            useExisting: forwardRef(() => MonthPickerComponent),
            multi: true
        }
    ]
})
export class MonthPickerComponent implements ControlValueAccessor {
```

**Expected Result**

I shouldn't get an error against `@typescript-eslint/no-use-before-define` on `useExisting` line because I'm explicitly telling the compiler that it's a `forwareRef`

**Actual Result**

Complains about using `MyComponent` before it's defined.

**Additional Info**

I know I can put the comment on that line to disable the check, but I shouldn't have to do that for this normal use case.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.30.0` |
| `@typescript-eslint/parser`        | `2.30.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `v14.0.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
See https://github.com/typescript-eslint/typescript-eslint/issues/1856#issuecomment-617879334
